L1Trigger/TrackFindingTracklet/plugins/ProducerKFout.cc

```diff
@@ -216,60 +252,88 @@ namespace trklet {
+          TTBV Z0(
+              temp_z0, dataFormats_->base(Variable::zT, Process::kf), TTTrack_TrackWord::TrackBitWidths::kZ0Size, true);
+          TTBV TanL(temp_tanL,
+                    dataFormats_->base(Variable::cot, Process::kf),
+                    TTTrack_TrackWord::TrackBitWidths::kTanlSize,
+                    true);
+          TTBV Phi0(temp_phi0,
+                    dataFormats_->base(Variable::phiT, Process::kf),
+                    TTTrack_TrackWord::TrackBitWidths::kPhiSize,
+                    true);
+          TTBV InvR(-InTrack.inv2R(),
+                    dataFormats_->base(Variable::inv2R, Process::kf),
+                    TTTrack_TrackWord::TrackBitWidths::kRinvSize + 1,
+                    true);
+          InvR.resize(TTTrack_TrackWord::TrackBitWidths::kRinvSize);
+
+          // Create input vector for BDT
+          trackQuality_inputs = {
+              (std::trunc(TanL.val() / tqTanlScale_)) / ap_fixed_rescale,
+              (std::trunc(Z0.val() / tqZ0Scale_)) / ap_fixed_rescale,
+              0,
+              temp_nstub,
+              temp_ninterior,
+              digitise(TTTrack_TrackWord::chi2RPhiBins, tempchi2rphi, (double)setup_->kfoutchi2rphiConv()),
+              digitise(TTTrack_TrackWord::chi2RZBins, tempchi2rz, (double)setup_->kfoutchi2rzConv())};
+
+          // Run BDT emulation and package output into 3 bits
+
+          tempTQMVA = trackQualityModel_->runEmulatedTQ(trackQuality_inputs);
+          tempTQMVA = std::trunc(tempTQMVA * ap_fixed_rescale);
+          TTBV TQMVA(digitise(tqBins_, tempTQMVA, 1.0), TTTrack_TrackWord::TrackBitWidths::kMVAQualitySize, false);
+
+          // Build 32 bit partial tracks for outputting in 64 bit packets
+          //                  12 +  3       +  7         +  3    +  6
           TTBV PartialTrack3((D0 + BendChi2 + HitPattern + TQMVA + extraMVA), partialTrackWordBits_, false);
-          // 16   + 12    + 4
-          TTBV PartialTrack2((TanL + z0 + Chi2rz), partialTrackWordBits_, false);
-          // 1        + 15   +  12 +    4
-          TTBV PartialTrack1((TrackValid + InvR + phi0 + Chi2rphi), partialTrackWordBits_, false);
+          //                  16   + 12    + 4
+          TTBV PartialTrack2((TanL + Z0 + Chi2rz), partialTrackWordBits_, false);
+          //                    1        + 15   +  12 +    4
+          TTBV PartialTrack1((TrackValid + InvR + Phi0 + Chi2rphi), partialTrackWordBits_, false);
 
           int sortKey = (InTrack.sectorEta() < (int)(setup_->numSectorsEta() / 2)) ? 0 : 1;
           // Set correct bit to valid for track valid
-          TrackKFOut Temp_track(
-              PartialTrack1.set(31), PartialTrack2, PartialTrack3, sortKey, track, iTrack, iLink, true);
-
-          InTrackStreams[iLink / numWorkers_][iLink % numWorkers_][iTrack] = (std::make_shared<TrackKFOut>(Temp_track));
-
+          TrackKFOut Temp_track(PartialTrack1.set((partialTrackWordBits_ - 1)),
```

Comment:
By CMS convention, all variable names, such as "Temp_track" and "InTrackStreams" must start with a lower case letter.